addons/portal_rating/static/src/js/portal_rating_composer.js

```diff
@@ -141,8 +141,10 @@ const RatingPopupComposer = publicWidget.Widget.extend({
             default_rating_value:
                 data.default_rating_value || this.rating_value || 4,
         };
-        Object.assign(data, defaultOptions);
-        this.options = Object.assign(this.options, data);
+        if (message && message.author && message.author.id === this.options.partner_id) {
```

Comment:
nullish is fine to use in non-negative conditions
```suggestion
        if (message?.author?.id === this.options.partner_id) {
```

Reply:
unless `partner_id` can be `undefined` and we need to ensure we don't get `undefined === undefined`?

Reply:
We could still at least simplify:
```suggestion
        if (message?.author && message.author.id === this.options.partner_id) {
```
or
```suggestion
        if (this.options.partner_id && message?.author?.id === this.options.partner_id) {
```